journal/src/main/java/io/camunda/zeebe/journal/file/SegmentWriter.java

```diff
@@ -234,8 +238,6 @@ private Either<SegmentFull, Integer> writeRecord(
     if (recordLength.isLeft()) {
       return Either.left(new SegmentFull("Not enough space to write record"));
     }
-    final int nextEntryOffset = offset + recordLength.get();
-    invalidateNextEntry(nextEntryOffset);
     return Either.right(recordLength.get());
```

Comment:
:wrench: We can map the `Either` returned directly, which is where its strength comes from :)

```java
return serializer.writeData(index, asqn, recordDataWriter, writeBuffer, offset)
        .mapLeft(e -> new SegmentFull("Not enough space to write record", e));
```